guess-the-country/country-frontend/src/components/Notification.vue

```diff
@@ -1,20 +1,31 @@
 <template>
   <div>
       <section v-show="!explanation" class="xd-section xd-light" >
-         <section v-show="!user_city_answer">
-            {{ msg }}
-         </section>   
-      <section v-show="user_city_answer && !prediction_city">
-        <p class="short-text">Your guess is: {{user_city_answer}}</p>
-      </section>
-      <section v-show="prediction_city && control">
-        <p class="short-text">My guess is: {{prediction_city}}</p>
-      </section>
-        <section v-show="prediction_city && !control">
-        <p>My guess is: {{prediction_city}}
-        <br>In particular, the colored areas below have helped me form my guess. </p>
-      </section>
-    
+        <!-- Question -->
+        <section v-show="sequence_mode==='classic'&&!user_city_answer
+                  || sequence_mode==='recommender'&&!prediction_city
+                  || sequence_mode==='basic'&&!user_city_answer">  
+          <p v-show= "sequence_mode==='classic'">Your guess: Where has this Google Streetview picture been taken?</p>
+          <p v-show= "sequence_mode==='basic'||sequence_mode==='recommender'">Where has this Google Streetview picture been taken?</p>
+        </section>   
+        <!-- User guess -->
+        <section v-show="sequence_mode==='classic'&&user_city_answer && !prediction_city
+                  || sequence_mode==='recommender'&&user_city_answer
+                  || sequence_mode==='basic' && user_city_answer">  
+          <p class="short-text">Your guess is: {{user_city_answer}}</p>
+        </section>
+        <!-- control group (without explanation), AI guess -->
+        <section v-show="sequence_mode==='classic'&&prediction_city && control
+                  || sequence_mode==='recommender'&& prediction_city&&!user_city_answer && control"> 
+          <p class="short-text">My guess is: {{prediction_city}}</p>
+        </section>
+        <!-- treatment group (with explanation), AI guess -->
+        <section v-show=" sequence_mode ==='classic' && prediction_city && !control
+                  || sequence_mode==='recommender' &&prediction_city&&!user_city_answer && !control"> 
```

Comment:
Die Berechnung, in welchem Modus wir sind, sollte wohl fertig berechnet als prop durchgereicht werden. Im Moment ist diese Logik an zahlreichen Stellen implementiert. Es wäre besser, das einmal in der `App.vue` zu berechnen und dann nur noch das Ergebnis durchzureichen.

Reply:
Ich hab alles in computed Methoden verpackt (siehe commit), ich versteh nicht ganz was du hier meinst? 

die Berechnung, welcher Modus aktiv ist passiert doch an der Stelle wo die env variable ausgelesen wird, und dieser Wert wird dann in der "sequence_mode" Variable in die Komponenten übergeben.
Und die Logik muss doch trotzdem überall stehen (jetzt halt im computed Bereich), weil die Logik ja für jedes Element unterschiedlich ist.

Oder hab ich da was übersehen?

Reply:
Alles klar, danke für die Erklärung! Für mich sah es so aus, als würde die identische Verkettung von Variablen mehrfach berechnet, aber es ist ja doch sehr unterschiedlich und der Teufel steckt im Detail. Umso wichtiger, dass wir das im Code möglichst klar und leserlich machen :-) Ich habe nochmal kommentiert und Beispiele gegeben, was ich damit konkret meine.